scikeras/utils/__init__.py

```diff
@@ -0,0 +1,114 @@
+from inspect import isclass
+from typing import Callable, Union
+
+import numpy as np
+
+from tensorflow.keras.losses import Loss
+from tensorflow.keras.losses import get as keras_loss_get
+from tensorflow.keras.metrics import Metric
+from tensorflow.keras.metrics import get as keras_metric_get
+
+
+def _camel2snake(s: str) -> str:
+    """from [1]
+    [1]:https://stackoverflow.com/questions/1175208/elegant-python-function-to-convert-camelcase-to-snake-case
+    """
+    return "".join(["_" + c.lower() if c.isupper() else c for c in s]).lstrip("_")
+
+
+def loss_name(loss: Union[str, Loss, Callable]) -> str:
+    """Retrieves a loss's full name (eg: "mean_squared_error").
+
+    Parameters
+    ----------
+    loss : Union[str, Loss, Callable]
+        Instance of Keras Loss, loss callable or string
+        shorthand (eg: "mse") or full name ("mean_squared_error").
+
+    Returns
+    -------
+    str
+        [description]
+
+    Notes
+    -----
+    The result of this function will always be in snake case, not camel case.
+
+    Examples
+    --------
+    >>> loss_name("BinaryCrossentropy")
+    'binary_crossentropy'
+    >>> loss_name("binary_crossentropy")
+    'binary_crossentropy'
+    >>> import tensorflow.keras.losses as losses
+    >>> loss_name(losses.BinaryCrossentropy)
+    'binary_crossentropy'
+    >>> loss_name(losses.binary_crossentropy)
+    'binary_crossentropy'
+
+    Raises
+    ------
+    TypeError
+        If the wrong type is passed.
+
+    """
+    if isclass(loss):
+        loss = loss()
+    if not (isinstance(loss, (str, Loss)) or callable(loss)):
+        raise TypeError(
```

Comment:
Why not remove this error and let TF handle errors?

``` python-traceback
[ins] In [18]: tf.keras.losses.get(["binary_crossentropy"])
---------------------------------------------------------------------------
ValueError                                Traceback (most recent call last)
<ipython-input-18-0632950f1cef> in <module>
----> 1 tf.keras.losses.get(["binary_crossentropy"])

~/anaconda3/lib/python3.7/site-packages/tensorflow/python/keras/losses.py in get(identifier)
   1900   else:
   1901     raise ValueError(
-> 1902         'Could not interpret loss function identifier: {}'.format(identifier))
   1903
   1904

ValueError: Could not interpret loss function identifier: ['binary_crossentropy']
```

Reply:
The errors get confusing, eg for a dict:
```python3
ValueError: Improper config format: {}
```
If we know that our function only works for bare inputs and we make that clear in the docstring, I think we should be raising a TypeError for other types of input instead of relying on TF to raise some cryptic ValueError.

Reply:
I see. I think a TypeError should still be raised.